pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1336 @@
+        if lowerBoundValue is not None:
+            if lowerBoundValue > 0:
+                raise InfeasibleConstraintException(
+                    'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+                )
+            elif lowerBoundValue == 0:
+                lowerBound = ' 0 = '
+            else:
+                lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ''
+
+        if upperBoundValue >= 0:
+            upperBound = ' \\leq 0 '
+        else:
+            upperBound = ' \\leq ' + str(upperBoundValue)
+
+    elif domainName in ['NegativeReals', 'NegativeIntegers']:
+        if lowerBoundValue is not None:
+            if lowerBoundValue >= 0:
+                raise InfeasibleConstraintException(
+                    'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+                )
+            else:
+                lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ''
+
+        if upperBoundValue >= 0:
+            upperBound = ' < 0 '
+        else:
+            upperBound = ' \\leq ' + str(upperBoundValue)
+
+    elif domainName in ['NonNegativeReals', 'NonNegativeIntegers']:
+        if lowerBoundValue > 0:
+            lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ' 0 \\leq '
+
+        if upperBoundValue is not None:
+            if upperBoundValue < 0:
+                raise InfeasibleConstraintException(
+                    'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+                )
+            elif upperBoundValue == 0:
+                upperBound = ' = 0 '
+            else:
+                upperBound = ' \\leq ' + str(upperBoundValue)
+        else:
+            upperBound = ''
+
+    elif domainName in ['Boolean', 'Binary', 'Any', 'AnyWithNone', 'EmptySet']:
+        lowerBound = ''
+        upperBound = ''
+
+    elif domainName in ['UnitInterval', 'PercentFraction']:
+        if lowerBoundValue > 1:
+            raise InfeasibleConstraintException(
+                'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+            )
+        elif lowerBoundValue == 1:
+            lowerBound = ' = 1 '
+        elif lowerBoundValue > 0:
+            lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ' 0 \\leq '
+
+        if upperBoundValue < 0:
+            raise InfeasibleConstraintException(
+                'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+            )
+        elif upperBoundValue == 0:
+            upperBound = ' = 0 '
+        elif upperBoundValue < 1:
+            upperBound = ' \\leq ' + str(upperBoundValue)
+        else:
+            upperBound = ' \\leq 1 '
+
+    else:
+        raise DeveloperError(
+            'Invalid domain somehow encountered, contact the developers'
+        )
```

Comment:
This is actually easy to hit.  Variable domains can be any valid Pyomo Set, so the following is perfectly legitimate:
```python
m = ConcreteModel()
m. v_domain = Set(initialize=[3,4,5,6])
m.x = Var(domain=m.v_domain)
```
The thing to use to infer domains are the Var `is_continuous()`, `is_integer()`, and `is_is_binary()` methods:
```python
>>> m.x.is_integer()
True
>>> m.x.is_binary()
False
>>> m.x.is_continuous()
False
>>> m.x.bounds
(3, 6)
```
For the LP / NL writers, we only require that one of those three return True.  So, for example, you can write a model with a domain `[1,2,3,7,8,9]`, but the solver interfaces will complain (because it is not a bounded contiguous integer domain)

Reply:
Changed this to a NotImplementedError for the time being